crates/db_queries/src/source/comment.rs

```diff
@@ -166,17 +166,17 @@ impl Likeable<CommentLikeForm> for CommentLike {
     use lemmy_db_schema::schema::comment_like::dsl::*;
     insert_into(comment_like)
       .values(comment_like_form)
-      .on_conflict((comment_id, user_id))
+      .on_conflict((comment_id, person_id))
       .do_update()
       .set(comment_like_form)
       .get_result::<Self>(conn)
   }
-  fn remove(conn: &PgConnection, user_id: i32, comment_id: i32) -> Result<usize, Error> {
+  fn remove(conn: &PgConnection, person_id: i32, comment_id: i32) -> Result<usize, Error> {
```

Comment:
I dont like that these methods simply use i32, its very easy to pass a `user_id` instead of `person_id`, or the other way round if you dont pay attention. I think we should add `struct PersonId(i32)` and `struct UserId(i32)`, used in db_schema structs, db_queries methods, and everywhere else these IDs are used. That way the compiler will complain if the wrong type of ID is passed in.

Reply:
I like that idea a lot.

Reply:
I've made a separate PR for this, because it was quite large, and affects a lot of things.